### PR TITLE
Deploy Improved logging 

### PR DESCRIPTION
**Pull request by jerryblakley**
@kibra Can you take a peek at a couple of places in the code, to double check my sanity and so someone else has an idea of what I'm doing here.

### CODE REVIEW

File: src/session.c [hunk 1]
```diff
@@ -768,16 +764,45 @@ CURL *session_request_init(const char *path, const char *query_string, bool temp
     // If the whole cluster is sad, avoid access altogether for a given period of time.
     // Calls to this function, on detecting this error, set ENETDOWN, which is appropriate
     if (use_saint_mode()) {
-        log_print(LOG_ERR, SECTION_SESSION_DEFAULT, "session_request_init: already in saint mode");
+        log_print(LOG_NOTICE, SECTION_SESSION_DEFAULT, "session_request_init: already in saint mode");
         return NULL;
     }
 
-    // If we add or delete a node, or change its health status, we need to signal here to
```
Comment (jerryblakley) [inline src/session.c]: @kibra Can you check this logic?

File: src/session.c [hunk 2]
```diff
@@ -877,6 +917,41 @@ static void increment_node_success(char *addr) {
     }
 }
 
+static void print_errors(const int iter, const char *type_str, const char *fcn_name,
```
Comment (jerryblakley) [inline src/session.c]: @kibra Can you check this logic?

File: src/session.c [hunk 3]
```diff
@@ -877,6 +917,41 @@ static void increment_node_success(char *addr) {
...
+
+    if (res != CURLE_OK) {
+        asprintf(&error_str, "%s :: %s", curl_easy_strerror(res), "no rc");
+    } else {
+        asprintf(&error_str, "%s :: %lu", "no curl error", response_code);
+    }
+
+    // Track number of failures
+    log_print(LOG_INFO, SECTION_ENHANCED,
+        "%s: curl iter %d on path %s; %s -- fusedav.%s.server-%s.%s:1|c",
```
Comment (kibra) [inline src/session.c]: `fusedav.%s.server-%s.by_iter.%s`

File: src/session.c [hunk 4]
```diff
@@ -877,6 +917,41 @@ static void increment_node_success(char *addr) {
...
+        fcn_name, iter, path, error_str, filesystem_cluster, nodeaddr, failure_str);
+
+    free(failure_str);
+
+    // Distinguish curl from 500-status failures
+    log_print(LOG_INFO, SECTION_ENHANCED,
+        "%s: curl iter %d on path %s; %s -- fusedav.%s.server-%s.%s:1|c",
```
Comment (kibra) [inline src/session.c]: `fusedav.%s.server-%s.by_type.%s`

**Comment by kibra** [comment]
+1

:ferris_wheel:

**Pull request merged**